sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStore.java

```diff
@@ -433,6 +440,92 @@ public void clearProperty(final String key) throws StateStoreException {
         }
     }
 
+    // Read/Write task metadata
+
+    /**
+     * Stores the goal state override status of a particular Task. The {@link TaskInfo} for this exact task MUST have
+     * already been written via {@link #storeTasks(Collection)} beforehand.
+     *
+     * @throws StateStoreException in the event of a storage error
+     */
+    public void storeGoalOverrideStatus(String taskName, GoalStateOverride.Status status)
+            throws StateStoreException {
+        try {
+            Map<String, byte[]> values = new TreeMap<>();
+
+            if (GoalStateOverride.Status.INACTIVE.equals(status)) {
+                // Mark inactive state by clearing any override bits.
+                values.put(getGoalOverridePath(taskName), null);
```

Comment:
What's more fault-tolerant do you think?  Attempting to keep `empty` meaning no overrides, or having an explicit value for no overrides.  `setter` and `getter` both have branches for present and not-present.  I wonder if we can simplify this by removing a branch.  It's definitely fine as is.

One other thing.  It seems a little strange to me to put `null` values for deleting a node.  Maybe the persister should have a `clearMany` method.

Reply:
Switched to having a new separate `deleteMany` method to be called here. Originally there had been a possibility of mixing writes and deletes in the same call, but that's not really needed now so having two separate calls makes things a lot simpler and/or less error-prone.